### PR TITLE
Update GPU workflows

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

Add new GPU workflows, that run the Patatrack pixel local and pixel-only track reconstruction in addition to the full reconstruction, with the possibility of offloading to GPUs also the ECAL and HCAL local reconstruction.

#### PR validation:

Validated with
```
$ runTheMatrix.py -w upgrade -j 4 -t 8 -l 11634.591,11634.592,11634.595,11634.596
...
Running up to 4 concurrent jobs, each with 8 threads per process
...
11634.591_TTbar_14TeV+2021_Patatrack_CPU+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Sat Sep 18 12:26:05 2021-date Sat Sep 18 12:21:24 2021; exit: 0 0 0 0
11634.592_TTbar_14TeV+2021_Patatrack_GPU+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Sat Sep 18 12:26:12 2021-date Sat Sep 18 12:21:25 2021; exit: 0 0 0 0
11634.595_TTbar_14TeV+2021_Patatrack_TripletsCPU+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Sat Sep 18 12:26:15 2021-date Sat Sep 18 12:21:25 2021; exit: 0 0 0 0
11634.596_TTbar_14TeV+2021_Patatrack_TripletsGPU+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Sat Sep 18 12:26:10 2021-date Sat Sep 18 12:21:26 2021; exit: 0 0 0 0
4 4 4 4 tests passed, 0 0 0 0 failed
```

Note: this PR includes the changes from #34978; it can be rebased once that is merged.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -554,6 +578,69 @@ def setup_(self, step, stepName, stepDict, k, properties):
     offset = 0.522,
 )
 
+upgradeWFs['PatatrackCPU'] = PatatrackWorkflow(
+    digi = {
+        # there is no customisation for enabling the Patatrack pixel quadruplets running only on the CPU
+    },
+    reco = {
+        # skip the @pixelTrackingOnlyValidation which cannot run together with the full reconstruction
```
Comment (fwyzard) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:586]: see https://hypernews.cern.ch/HyperNews/CMS/get/recoTracking/1939.html

Comment (mmusich) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:586]: @sroychow @arossi83 @tsusa @connorpa please take note of this issue triggered in the phase1 pixel validation code when using the pixelonly reconstruction (by the way if the issue is only in `SiPixelPhase1HitsV` I suppose one could cook temporarily a lighter ad hoc validation sequence, just removing that module).

**Comment by Martin-Grunewald** [comment]
+1

**Comment by qliphy** [comment]
@cms-sw/pdmv-l2  Any comment?

**Comment by kskovpen** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**